src/main/Menu.java

```diff
@@ -0,0 +1,7 @@
+package main;
+
+import java.util.Iterator;
+
+public interface Menu {
+    Iterator createIterator();
```

Comment:
In this case I think `createIterator` could be given a better name, like `createMenuItemIterator`.